lib/Parse/ParseSIL.cpp

```diff
@@ -683,6 +689,80 @@ static bool parseSILOptional(bool &Result, SILParser &SP, StringRef Expected) {
   return false;
 }
 
+namespace {
+  /// A helper class to perform lookup of IdentTypes in the
+  /// current parser scope.
+  class IdentTypeReprLookup : public ASTWalker {
```

Comment:
IMHO, it has very little to do with layout requirements. It is required for the lookup of the generic type parameters of the current SIL function. See my comment above.